websockets/bridge.go

```diff
@@ -0,0 +1,116 @@
+package websockets
+
+import (
+	"fmt"
+
+	"github.com/gorilla/websocket"
+	"github.com/pokt-network/poktroll/pkg/polylog"
+)
+
+// bridge routes data between an Endpoint and a Client. One bridge represents
+// a single WebSocket connection between a Client and a WebSocket Endpoint.
+//
+// Full data flow: Client <------> PATH <------> WebSocket Endpoint
+type bridge struct {
+	logger       polylog.Logger
+	endpointConn *connection
+	clientConn   *connection
+	msgChan      <-chan message
+	stopChan     chan error
+}
+
+// NewBridge creates a new Bridge instance and a new connection to the Endpoint from the Endpoint URL
+func NewBridge(logger polylog.Logger, endpointURL string, clientWSSConn *websocket.Conn) (*bridge, error) {
+	endpointWSSConn, err := connectEndpoint(endpointURL)
+	if err != nil {
+		return nil, fmt.Errorf("error establishing connection to endpoint URL %s: %s", endpointURL, err.Error())
+	}
+
+	msgChan := make(chan message)
+	stopChan := make(chan error)
+
+	logger = logger.With("component", "bridge")
+
+	endpointConnection := newConnection(
+		logger.With("conn", "endpoint"),
+		endpointWSSConn,
+		messageSourceEndpoint,
+		msgChan,
+		stopChan,
+	)
+	clientConnection := newConnection(
+		logger.With("conn", "client"),
+		clientWSSConn,
+		messageSourceClient,
+		msgChan,
+		stopChan,
+	)
+
+	return &bridge{
+		endpointConn: endpointConnection,
+		clientConn:   clientConnection,
+		msgChan:      msgChan,
+		stopChan:     stopChan,
+
+		logger: logger,
+	}, nil
+}
+
+/* ---------- Public method - Run Bridge ---------- */
```

Comment:
Should w create an interface and enforce adherence so it's self explanatory?

Reply:
Not sure if we need to since the `bridge` struct is only used in one possible way. My preference is not to define an interface here unless it's actually needed. That said, the comment is superfluous so I will remove it.